src/IdObfuscator/IdObfuscator.php

```diff
@@ -0,0 +1,69 @@
+<?php
+namespace Craftwork\IdObfuscator;
+
+final class IdObfuscator
+{
+    const CHARSET_DEFAULT = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-_';
+
+    /**
+     * @var string
+     */
+    private $characterSet;
+
+    /**
+     * @var int
+     */
+    private $characterSetLength;
+
+    /**
+     * @param string $salt Salt that should be unique and kept secret for your application.
+     * @param string $characterSet An optional character set.
+     */
+    public function __construct(string $salt, string $characterSet = self::CHARSET_DEFAULT)
+    {
+        $this->characterSet = $this->shuffleCharacterSetWithSalt($characterSet, $salt);
+        $this->characterSetLength = strlen($this->characterSet);
+    }
+
+    /**
+     * @param int $number The number to encode.
+     * @return string
+     */
+    public function encode(int $number) : string
+    {
+        $hash = '';
+
+        do {
+            $hash = $this->characterSet[$number % $this->characterSetLength] . $hash;
+            $number = (int) ($number / $this->characterSetLength);
+        } while ($number > 0);
+
+        return $hash;
+    }
+
+    public function decode(string $string) : int
+    {
+        return array_reduce(str_split($string), function ($number, $char) {
+            return ($number * $this->characterSetLength) + strpos($this->characterSet, $char);
+        }, 0);
+    }
+
+    private function shuffleCharacterSetWithSalt(string $characterSet, string $salt): string
+    {
+        $saltLength = strlen($salt);
+        $shuffled = $characterSet;
+
+        if ($saltLength === 0) {
+            return $shuffled;
+        }
+
+        for ($i = strlen($characterSet) - 1, $v = 0, $p = 0; $i > 0; $i--, $v++) {
+            $v %= $saltLength;
+            $p += $int = ord($salt[$v]);
```

Comment:
I don't know what any of the variables should be called actually. Even 'int' is unclear. Anyone got any suggestions?